### PR TITLE
New msgpack extensions, extended connection options, bug fixes

### PR DESCRIPTION
**Pull request by goodwise**
- Added 3 new msgpack extensions: UUID, Datetime, Decimal.
- Connection object now accepts all options of `net.createConnection()`, including Unix socket path.
- New `nonWritableHostPolicy` and related options, which improves a high availability capabilities without any 3rd parties.
- Ability to disable the offline queue.
- Fixed [bug with int32](https://github.com/tarantool/node-tarantool-driver/issues/48) numbers when it was encoded as floating. Use method `packInteger()` to solve this.
- `selectCb()` now also accepts `spaceId` and `indexId` as their String names, not only their IDs.
- Some performance improvements by caching internal values.
- TLS (SSL) support.
- New `pipeline()`+`exec()` methods kindly borrowed from the [ioredis](https://github.com/redis/ioredis?tab=readme-ov-file#pipelining), which lets you to queue some commands in memory and then send them simultaneously to the server in a single (or several, if request body is too big) network call(s). Thanks to the Tarantool, which [made this possible](https://www.tarantool.io/en/doc/latest/dev_guide/internals/iproto/format/#packet-structure).
This way the performance is significantly improved by 500-1600% - you can check it yourself by running `npm run benchmark-read` or `npm run benchmark-write`.
Note that this feature doesn't replaces the Transaction model, which has some level of isolation.
- Changed `const` declaration to `var` in order to support old Node.JS versions.

### CODE REVIEW

**Comment by sergos** [comment]
Sorry, but `test.sh` broken exactly at the first commit.
Apparently, `istanbul` isn't aware of `?.` and fails w/some sort of parsing.

**Comment by goodwise** [comment]
Will inspect this.

What is the reason to use the Denque? The whole functionality of this module could be implemented with a native Array's methods like push / pop / shift / splice, which are 2 times faster in my own benchmarks and available since node 0.10.0 version

**Comment by goodwise** [comment]
> is broken

Fixed it

**Comment by goodwise** [comment]
@sergos @oleg-jukovec any news?

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Node tarantool driver for 1.7+ support Node.js v.4+.
 
 Based on [go-tarantool](https://github.com/tarantool/go-tarantool) and implements [Tarantool’s binary protocol](http://tarantool.org/doc/dev_guide/box-protocol.html), for more information you can read them or basic documentation at [Tarantool manual](http://tarantool.org/doc/).
 
-Code architecture and some features in version 3 borrowed from the [ioredis](https://github.com/luin/ioredis).
+Code architecture and some features in version 3 borrowed from the [ioTarantool](https://github.com/luin/ioTarantool).
```
Comment (filonenko-mikhail) [inline README.md:9]:
```suggestion
Code architecture and some features in version 3 borrowed from the [ioredis](https://github.com/luin/ioredis).
```

File: README.md [hunk 2]
```diff
@@ -85,7 +93,7 @@ except when the connection is closed manually by `tarantool.disconnect()`.
 It's very flexible to control how long to wait to reconnect after disconnection
 using the `retryStrategy` option:
 
-```javascript
+```Javascript
```
Comment (filonenko-mikhail) [inline README.md:96]:
```suggestion
```javascript
```

File: README.md [hunk 3]
```diff
@@ -102,12 +110,12 @@ the return value represents how long (in ms) to wait to reconnect. When the
 return value isn't a number, node-tarantool-driver will stop trying to reconnect, and the connection
 will be lost forever if the user doesn't call `tarantool.connect()` manually.
 
-**This feature is borrowed from the [ioredis](https://github.com/luin/ioredis)**
+**This feature is borrowed from the [ioTarantool](https://github.com/luin/ioTarantool)**
```
Comment (filonenko-mikhail) [inline README.md:113]:
```suggestion
**This feature is borrowed from the [ioredis](https://github.com/luin/ioredis)**
```

**Review by filonenko-mikhail — APPROVED** [review]

**Comment by sergos** [comment]
> > is broken
> 
> Fixed it

As of today tests are passing, no more questions from me.

**Review by sergos — APPROVED** [review]
LGTM

**Pull request merged**